connector-builder-server/src/main/openapi/openapi.yaml

```diff
@@ -0,0 +1,246 @@
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/StreamReadRequestBody"
+        required: true
+      responses:
+        "200":
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/StreamRead"
+        "400":
+          $ref: "#/components/responses/ExceptionResponse"
+        "422":
+          $ref: "#/components/responses/InvalidInputResponse"
+  /v1/streams/list:
+    post:
+      summary: List all streams present in the connector definition, along with their specific request URLs
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/StreamsListRequestBody"
+        required: true
+      responses:
+        "200":
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/StreamsListRead"
+        "400":
+          $ref: "#/components/responses/ExceptionResponse"
+        "422":
+          $ref: "#/components/responses/InvalidInputResponse"
+
+components:
+  schemas:
+    StreamRead:
+      type: object
+      required:
+        - slices
+      properties:
+        slices:
+          type: array
+          description: The stream slices returned from the read command
+          items:
+            type: object
+            required:
+              - sliceId
+              - pages
+            properties:
+              sliceId:
+                type: string
+                description: String identifying the current slice
+              pages:
+                type: array
+                description: The pages returned from the read command
+                items:
+                  type: object
+                  required:
+                    - pageId
+                    - results
+                    - logs
+                    - request
+                    - response
+                  properties:
+                    pageId:
+                      type: string
+                      description: String identifying the current page
+                    results:
+                      type: array
+                      description: The RECORD and STATE AirbyteMessages coming from the read operation for this page
+                      items:
+                        type: string
+                    logs:
+                      type: array
+                      description: The LOG AirbyteMessages coming from the read operation for this page
+                      items:
+                        type: string
+                    request:
+                      type: object
+                      description: The HTTP request sent to the source API for this page
+                      required:
+                        - url
+                        - headers
```

Comment:
I'm not 100% sure if headers should be required or not, but it should be consistent with the body and request parameters fields